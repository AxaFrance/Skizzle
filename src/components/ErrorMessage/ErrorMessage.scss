.skz-error-message {
	position: absolute;
	left: 1rem;
	right: 1rem;
	top: 50%;
	transform: translateY(-50%);
	text-align: center;

	p {
		margin-bottom: 0.5rem;
	}

	button {
		font-size: 0.8rem;
		text-decoration: underline;
		cursor: pointer;
		border: 0;
		background-color: transparent;

		&:hover {
			text-decoration: none;
		}
	}

	@media (prefers-color-scheme: dark) {
		color: #fff;

		button {
			color: #ccc;
		}
	}
}
