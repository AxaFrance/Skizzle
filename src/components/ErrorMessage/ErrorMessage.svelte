<script>
	export let retry;
	export let label;
</script>

<style src="./ErrorMessage.scss">

</style>

<div class="skz-error-message">
	<p>{label}</p>
	<button on:click={retry}>RÃ©essayer</button>
</div>
