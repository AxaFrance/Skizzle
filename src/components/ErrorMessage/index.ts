export { default } from './ErrorMessage.svelte';
