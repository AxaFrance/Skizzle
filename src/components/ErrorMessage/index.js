import ErrorMessage from './ErrorMessage.svelte';

export default ErrorMessage;
