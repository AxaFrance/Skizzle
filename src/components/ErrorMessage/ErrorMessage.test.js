import { render, fireEvent } from '@testing-library/svelte';
import ErrorMessage from './ErrorMessage.svelte';

describe('Testing Error Message Component', () => {
	it('should render with empty props', () => {
		const { container } = render(ErrorMessage, {
			label: '',
			retry: jest.fn(),
		});

		expect(container).toMatchSnapshot();
	});

	it('should render with label', () => {
		const { container } = render(ErrorMessage, {
			label: 'An error occurred!',
			retry: jest.fn(),
		});

		expect(container).toMatchSnapshot();
	});

	it('should render with button', async () => {
		const retry = jest.fn();
		const { getByText } = render(ErrorMessage, {
			label: '',
			retry,
		});

		const button = getByText('Retry');

		await fireEvent.click(button);

		expect(retry).toHaveBeenCalled();
	});
});
