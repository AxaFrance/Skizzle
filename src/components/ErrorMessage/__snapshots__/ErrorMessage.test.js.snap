// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Error Message Component should render with empty props 1`] = `
<body>
  <div>
    <div
      class="skz-error-message"
    >
      <p>
        
      </p>
       
      <button>
        Réessayer
      </button>
    </div>
  </div>
</body>
`;

exports[`Testing Error Message Component should render with label 1`] = `
<body>
  <div>
    <div
      class="skz-error-message"
    >
      <p>
        An error occurred!
      </p>
       
      <button>
        Réessayer
      </button>
    </div>
  </div>
</body>
`;
